crates/services/relayer/src/service/state.rs

```diff
@@ -53,50 +49,34 @@ impl EthState {
     }
 
     pub fn is_synced_at(&self) -> Option<u64> {
-        self.local.filter(|local| *local >= self.remote.finalized())
+        self.local.filter(|local| *local >= self.remote)
     }
 
     /// Get the gap between the relayer and the Ethereum node if
     /// a sync is required.
     pub fn needs_to_sync_eth(&self) -> Option<EthSyncGap> {
         (!self.is_synced()).then(|| {
-            EthSyncGap::new(
-                self.local.map(|l| l.saturating_add(1)).unwrap_or(0),
-                self.remote.finalized(),
-            )
+            let local = self.local.map(|l| l.saturating_add(1)).unwrap_or(0);
```

Comment:
Does removing the use of an inclusive range affect the correctness of doing a saturating_add here? It seems like there's a potential that local could be higher than remote in this case.

Reply:
I do not believe that this changes the correctness.  Prior to these changes, `EthSyncGap` used a single inclusive range to store its start and end values (`local` and `remote` respectively); with these changes, `EthSyncGap` stores its start and end values separately, and implements a conversion into an inclusive range where any range behaviour needs to be maintained. In effect, the correctness will be maintained.

At the point of instantiation, where we calculate the inputs to the `EthSyncGap` constructor, `local` remains unchanged, and `remote` is now the value returned by the Ethereum client. Before and after these changes, there is no guarantee that `local` was less than or equal to `remote`. This means that the `EthSyncGap` can be "empty", i.e., `local` is greater than or equal to `remote`. In fact, some tests check the behaviour when `local` is greater than `remote` to ensure that no sync is necessary: https://github.com/FuelLabs/fuel-core/blob/ec6a61277118d2b9e3deb9e29ae01f8129242ea8/crates/services/relayer/src/service/state/test.rs#L30